### PR TITLE
Add a script to upload documentation to Hackage

### PR DESCRIPTION
**Pull request by gromakovsky**
The script is supposed to be self-descriptive, I have explained what it does, how to use and why it's needed.

### CODE REVIEW

File: haskell/upload-hackage-docs.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+#
+# SPDX-License-Identifier: MPL-2.0
```
Comment (gromakovsky) [inline haskell/upload-hackage-docs.sh:5]: @maksbotan I don't know about the origin of this script so please check whether you are ok with this copyright and license.

File: haskell/upload-hackage-docs.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+dist=$(stack path --dist-dir 2>/dev/null)
+version=$(stack query locals "${PACKAGE_NAME}" version 2>/dev/null)
+cd "${PACKAGE_DIR}/${dist}/doc/html"
+# Hackage expects documentation in dir with -docs suffix.
+cp -r "${PACKAGE_NAME}" "${PACKAGE_NAME}-${version}-docs"
+# Hackage does not like GNU tar's default format, have to use "ustar".
+tar cvz --format=ustar -f "${PACKAGE_NAME}.tar.gz" "${PACKAGE_NAME}-${version}-docs"
+
+# Alternatively instead of passwords use
+# -H 'Authorization: X-ApiKey TOKEN'
```
Comment (balsoft) [inline haskell/upload-hackage-docs.sh]: Maybe add this ability to the script? Check if `HACKAGE_TOKEN` is defined and use it, and only if it's not, resort to passwords.

Comment (gromakovsky) [reply haskell/upload-hackage-docs.sh]: Oh god, I somehow unexpectedly appeared in the situation when I have to write some bash (other than adding comments and quotes) :oh-god-why:

Okay, I will do that (hopefully).

Comment (maksbotan) [reply haskell/upload-hackage-docs.sh]: Maybe run shellcheck afterwards just in case :p

File: haskell/upload-hackage-docs.sh (hunk 1 above)
Comment (maksbotan) [reply haskell/upload-hackage-docs.sh:5]: I think that I based it on https://github.com/phaazon/stack-haddock-upload. It's "BSD 0-Clause" license, so technically we are free to do whatever we want without the need to mention that repo.

And maybe there was some stuff on StackOverflow I've also used, but I can't find that now.

File: haskell/upload-hackage-docs.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+#
+# SPDX-License-Identifier: MPL-2.0
+
+# Authors: @maksbotan, @gromak
+
+# Use this script if you want to manually upload documentation to Hackage.
+# Sometimes Hackage fails to build your package and you don't want to fix
+# it but want to make your documentation available.
+# You need to specify:
+# • ${PACKAGE_NAME} is a name of a package you are uploading docs for
```
Comment (maksbotan) [inline haskell/upload-hackage-docs.sh:13]: BTW, maybe fail if not specified?

File: haskell/upload-hackage-docs.sh (hunk 2 above)
Comment (gromakovsky) [reply haskell/upload-hackage-docs.sh:34]: I have `set -u` in the beginning.

File: haskell/upload-hackage-docs.sh (hunk 3 above)
Comment (gromakovsky) [reply haskell/upload-hackage-docs.sh:13]: I have `set -u` in the beginning.

File: haskell/upload-hackage-docs.sh (hunk 2 above)
Comment (gromakovsky) [reply haskell/upload-hackage-docs.sh]: Okay, done. I have tested it and it even seems to work (both scenarios) :pogchamp:

Comment (gromakovsky) [reply haskell/upload-hackage-docs.sh]: Btw, it would be nice if CI in this repo applied shellcheck to all bash scripts automatically :pled:

**Comment by gromakovsky** [comment]
@balsoft @maksbotan :zhdun:

**Review by maksbotan — APPROVED** [review]

**Pull request merged**